### PR TITLE
chore: use OCaml 5.1.1-rc1

### PR DESCRIPTION
**Pull request by anmonteiro**

### CODE REVIEW

**Comment by anmonteiro** [comment]
@jchavarri please make sure setting up a new switch still works.

**Review by jchavarri — APPROVED** [review]
Everything seems to work fine.

Installation would be faster if instead of pinning packages manually from make command, we used `pin-depends`. Doing it manually means that opam will install each pinned package, then rebuild its deps. This can lead to some packages being rebuilt multiple times throughout the process. If you're ok, I can do this in a separate PR.

**Comment by jchavarri** [comment]
Went ahead and shared the proposal in https://github.com/melange-re/melange/pull/962, lmk what you think!

File: .github/workflows/opam-build.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
           - ubuntu-latest
           # - windows-latest # for some reason windows build is not picking dune 3.5
         ocaml-compiler:
-          - ocaml-variants.5.1.1+trunk
+          - ocaml-base-compiler.5.1.1~rc1
```
[the remark this answers is not in the document]
Comment (anmonteiro) [reply .github/workflows/opam-build.yml:28]: - I believe `5.1.1+trunk` grabs the latest commit from the 5.1 branch in github:ocaml/ocaml
- no idea if we can just use `5.1.1~rc1`. I took this incantation from [setup-ocaml](https://github.com/ocaml/setup-ocaml)
    - opam is usually so finicky that when something works I don't touch it anymore

**Pull request merged**